realm/realm-library/src/main/cpp/io_realm_SyncManager.cpp

```diff
@@ -93,6 +78,12 @@ JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeConfigureMetaDataSystem(J
     try {
         JStringAccessor base_file_path(env, baseFile); // throws
         SyncManager::shared().configure_file_system(base_file_path, SyncManager::MetadataMode::NoEncryption);
+
+        // Register Sync Client thread start/stop callback
+        g_binding_callback_thread_observer = &s_client_thread_listener;
```

Comment:
It looks a bit weird that this setup is in a method named `configureMetaDataSystem`. Perhaps rename this method to `initializeSync` or `initializeSyncManager` ?